Semantics.md

```diff
@@ -330,54 +321,67 @@ global variables will be necessary to allow sharing reference types between
+All constructs are formed out of the following control instructions:
 
- * `nop`: an empty operator that does not yield a value 
- * `block`: a fixed-length sequence of expressions with a label at the end
- * `loop`: a block with an additional label at the beginning which may be used to form loops
- * `if`: if expression with a list of *then* expressions and a list of *else* expressions
+ * `nop`: no operation, no effect
+ * `block`: the beginning of a block construct, a sequence of instructions with a label at the end
+ * `loop`: a block with a label at the beginning which may be used to form loops
+ * `if`: the beginning of an if construct with an implicit *then* block
+ * `else`: marks the else block of an if
  * `br`: branch to a given label in an enclosing construct
  * `br_if`: conditionally branch to a given label in an enclosing construct
  * `br_table`: a jump table which jumps to a label in an enclosing construct
  * `return`: return zero or more values from this function
+ * `end`: an instruction that marks the end of a block, loop, if, or function
+
+Blocks are composed of matched pairs of `block` ... `end` instructions, loops with matched pairs of 
+`loop` ... `end` instructions, and ifs with either `if` ... `end` or `if` ... `else` ... `end` sequences.
+For each of these constructs the instructions in the ellipsis are said to be *enclosed* in the
+construct.
 
 ### Branches and nesting
 
-The `br` and `br_if` constructs express low-level branching.
-Branches may only reference labels defined by an outer *enclosing construct*,
-which can be a `block` (with a label at the `end`), `loop` (with a label at the
-beginning), `if` (with a label at the `end` or `else`), `else` (with a label at
-the `end`), or the function body (with a label at the `end`). This means that,
-for example, references to a `block`'s label can only occur within the
-`block`'s body.
+The `br`, `br_if`, and `br_table` instructions express low-level branching and are hereafter refered to simply as branches.
+Branches may only reference labels defined by a construct in which they are enclosed.
+For example, references to a `block`'s label can only occur within the `block`'s body.
 
 In practice, outer `block`s can be used to place labels for any given branching
-pattern, except for one restriction: one can't branch into the middle of a loop
-from outside it. This restriction ensures all control flow graphs are well-structured
-in the exact sense as in high-level languages like Java, JavaScript, Rust and Go. To
-further see the parallel, note that a `br` to a `block`'s label is functionally
-equivalent to a labeled `break` in high-level languages in that a `br` simply
-breaks out of a `block`.
+pattern, except that the nesting restriction makes it impossible to branch into the middle of a loop
+from outside the loop. This limitation ensures by construction that all control flow graphs
+are well-structured as in high-level languages like Java, JavaScript, Rust and Go.
```

Comment:
Nit: Not sure if I'd call Rust a high-level language. Maybe a more relative "higher-level" is better?